### PR TITLE
Allow for extended function semantics in iffy

### PR DESCRIPTION
**Pull request by eprykhodko**
Addressed issue #63

### CODE REVIEW

File: funcy/funcs.py [hunk 1]
```diff
@@ -94,6 +93,7 @@ def iffy(pred, action=EMPTY, default=identity):
     if action is EMPTY:
         return iffy(bool, pred, default)
```
Comment (rozenbom) [inline funcy/funcs.py]: This will fail for `iffy({1,2,3})`. Why don't put `pred = make_pred(pred)` at the beginning of the function?

Comment (eprykhodko) [reply funcy/funcs.py]: Semantics of calling iffy with one argument implies that that argument is `action`, which must be callable.  According to the documentation the notion of extended function semantics is only applicable to predicates. `iffy({1,2,3})` should fail.

Comment (rozenbom) [reply funcy/funcs.py]: But doesn't "extended function semantics" imply converting sets to predicates (meaning something callable)?

Comment (eprykhodko) [reply funcy/funcs.py]: @rozenbom EFN is not about conversing sets to callables, it's about extending what a predicate can be. When you call iffy with one arg, that arg is not predicate - it's action. `iffy(x)` is a shorthand for `iffy(pred=bool, action=x)`

Comment (Suor) [reply funcy/funcs.py]: Actually extended semantics apply to mapper functions too, so in one argument `iffy()` argument should be passed to `make_func()` to get it.

**Comment by Suor** [comment]
Merged, thanks to everyone.

**Comment by eprykhodko** [comment]
Thank you!

**Pull request merged**